protocol/x/subaccounts/types/subaccount.go

```diff
@@ -73,26 +73,25 @@ func (m *Subaccount) GetUsdcPosition() *big.Int {
 }
 
 // SetUsdcAssetPosition sets the balance of the USDC asset position to `newUsdcPosition`.
-// If the absolute value of `newUsdcPosition` cannot be represented in a uint64,
-// an error is returned.
-func (m *Subaccount) SetUsdcAssetPosition(newUsdcPosition *big.Int) error {
-	if m != nil {
-		usdcAssetPosition := m.getUsdcAssetPosition()
-		if newUsdcPosition == nil || newUsdcPosition.Sign() == 0 {
-			if usdcAssetPosition != nil {
-				m.AssetPositions = m.AssetPositions[1:]
-			}
-		} else {
-			if usdcAssetPosition == nil {
-				usdcAssetPosition = &AssetPosition{
-					AssetId: assettypes.AssetUsdc.Id,
-				}
-				m.AssetPositions = append([]*AssetPosition{usdcAssetPosition}, m.AssetPositions...)
+func (m *Subaccount) SetUsdcAssetPosition(newUsdcPosition *big.Int) {
```

Comment:
This function also doesn't return an error anymore, so removed it + refactored it to early return.

Let me know if anyone would prefer I remove the refactor logic to early return. This function has good test coverage though so I felt like the change is safe.